src/aof.c

```diff
@@ -788,8 +811,17 @@ int openNewIncrAofForAppend(void) {
     /* Reset the aof_last_incr_size. */
     server.aof_last_incr_size = 0;
     /* Update `server.aof_manifest`. */
-    aofManifestFreeAndUpdate(temp_am);
+    if (temp_am) aofManifestFreeAndUpdate(temp_am);
     return C_OK;
+
+cleanup:
+    if (newfd != -1) close(newfd);
+    if (temp_am) {
+        aofManifestFree(temp_am);
+    } else {
+        sdsfree(new_aof_name);
```

Comment:
maybe instead of an else here, we should just always call sdsfree, and make sure to nullify the var on declaration and after freeing it elsewhere. seems easier to follow.